### PR TITLE
Todo list: add remove and edit functionality

### PR DESCRIPTION
**Pull request by ismayito**
#### Summary;
- Add todo list from the input
- Populate a list of the todo items from an array in the local storage into the list section dynamically
- Design the list to appear like the one in the given design
- Add ,edit , delete, and update the todo list items

### CODE REVIEW

File: src/modules/add.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    this.isCompleted = isCompleted;
+  }
+
+  const inputField = document.querySelector('.todo-input');
+
+  inputField.addEventListener('keyup', (event) => {
+    if (event.key === 'Enter') {
+      const listArray = retrieveFromStore();
+      const inputValue = inputField.value;
+      const index = listArray.length;
```
Comment (shahadat3669) [inline src/modules/add.js:19]: By default, newly added tasks possess the property `completed` set to false, which is a good start. However, there's a slight adjustment required:

- [ ] According to the project specifications, new tasks should also be equipped with the property `index`, set to the value of the new array length. Presently, if the third task is appended to the list, its index should be 3. However, in your current implementation, the index value of the 3rd task is 2. I recommend revising your code to align with the project's requirements and ensure accurate indexing.
image

File: src/modules/displayList.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const listItem = document.createElement('div');
+    listItem.classList.add('list-div');
+    const todoItem = document.createElement('input');
+    todoItem.type = 'text';
+    todoItem.classList.add('list-item');
+    todoItem.setAttribute('id', `input-${item.index}`);
+    const checkbox = document.createElement('input');
+    checkbox.classList.add('checkbox-item');
+    checkbox.type = 'checkbox';
+    checkbox.setAttribute('id', `${item.index}`);
```
Comment (shahadat3669) [inline src/modules/displayList.js:22]: Well done on incorporating dynamic `id` attributes for each checkbox, which is a positive step. However, after reviewing the page, I noticed an area for improvement:

- [ ] [Optional] When an item has a `completed` property value of `true`, the corresponding list item should be checked upon reloading the page. I recommend enhancing your code to dynamically assign the status of the checkbox based on the `completed` property, similar to how you've implemented the dynamic `id` attribute. This adjustment will ensure that the checkboxes accurately reflect the completion status of the tasks.

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @ismayito, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Has a function for adding a new task.
✅ Has a function for editing task descriptions.
✅ Has a function for deleting a task.
✅ Deleting a task and updating all remaining items' indexes.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @ismayito 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ "Add to-do" functionality 📌 
✔️ "Remove to-do" functionality 📌 
✔️ "Editing to-do" functionality 📌 
✔️ No linter errors 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**